firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/Firestore.kt

```diff
@@ -162,3 +172,67 @@ class FirebaseFirestoreKtxRegistrar : ComponentRegistrar {
     override fun getComponents(): List<Component<*>> =
             listOf(LibraryVersionComponent.create(LIBRARY_NAME, BuildConfig.VERSION_NAME))
 }
+
+/**
+ * Transforms a [DocumentReference] into a coroutine [Flow]
+ *
+ * **Backpressure handling**: by default this method conflates items. If the consumer isn't fast enough,
+ * it might miss some values but is always guaranteed to get the latest value. Use [bufferCapacity]
+ * to change that behaviour
+ *
+ * @param metadataChanges controls metadata-only changes. Default: [MetadataChanges.EXCLUDE]
+ * @param bufferCapacity the buffer capacity as in [Flow.buffer] or null to not buffer at all
+ */
+fun DocumentReference.toFlow(
```

Comment:
```suggestion
fun DocumentReference.documentSnapshots(
```

Reply:
I went with just `snapshots()` in [this commit](https://github.com/firebase/firebase-android-sdk/pull/1252/commits/288ae9aed3ea4691aac5efab5bcbdddd999e5af8). As @svenjacobs [said](https://github.com/firebase/firebase-android-sdk/pull/1252#discussion_r938064128), `documentSnapshots()` is more verbose and the "document" part is already in the "documentRef". Let me know if you want me to change it or feel free to amend.

Reply:
Sounds good to me.